feat: add initial storage automatically (#442)
FuelLabs/fuels-rs: docs/src/getting-started/basics.md

{{#include ../../../examples/contracts/src/lib.rs:deploy_with_parameters}}
```

### Initializing storage slots manually

The storage slots of a contract can be initialized manually from `Vec<StorageSlot>` where `StorageSlot` is a struct that holds the key-value pair for a given slot.

```rust,ignore
{{#include ../../../packages/fuels-abigen-macro/tests/harness.rs:storage_slot_create}}
```
Once created, the slots can be used to initialize `StorageConfiguration::with_manual_storage` which was then passed to `deploy_with_parameters`: 

```rust,ignore
{{#include ../../../packages/fuels-abigen-macro/tests/harness.rs:manual_storage}}
```

### Initializing storage slots automatically

> The storage slots of a contract can be initialized automatically from auto-generated JSON file where literals containing key-value pair are transformed into `Vec<StorageSlot>`.

Reviewer: IMO we should give the user more context on where the JSON file is coming from. We could also omit the details of how the read values are transformed. Maybe something like this?

```suggestion
The storage slots of a contract can be initialized automatically from the JSON file generated by the sway compiler.
```

Path to JSON file is used to initialize `StorageConfiguration::with_storage_path` which was then passed to `deploy_with_parameters`:

```rust,ignore
{{#include ../../../packages/fuels-abigen-macro/tests/harness.rs:automatic_storage}}
```

### Initializing storage slots both automatically and manually

> If we initialize the slots both automatically and manually, then the generated `Vec<StorageSlot>` will be merged and the `manually` generated `Vec<StorageSlot>` takes precedence.

Reviewer: IIUC this requires the user to use `StorageConfiguration::new()`. I think we should demonstrate this or at least mention how this can be achieved
